autoscoper/src/ui/AutoscoperMainWindow.cpp

```diff
@@ -952,13 +953,28 @@ void AutoscoperMainWindow::save_tracking_results(QString filename)
     bool interpolate = diag->diag->radioButton_InterpolationSpline->isChecked();
     int volume = -1;
     if (diag->diag->radioButton_VolumeCurrent->isChecked()) volume = tracker->trial()->current_volume;
+    exportLogFile();
     save_tracking_results(filename, save_as_matrix,save_as_rows,save_with_commas,convert_to_cm,convert_to_rad,interpolate, volume);
-
     is_tracking_saved = true;
   }
   delete diag;
 }
 
+void AutoscoperMainWindow::exportLogFile() {
+  xromm::OptimizationParameters& opt_param = tracker->getLatestOptimizationParameters();
+  QString filename = default_root_path + QString(OS_SEP) + "Logs" + QString(OS_SEP) + QString::number(std::time(nullptr)) + "_" + default_task_name + ".log";
+  std::cout << "Exporting log file to: " << filename.toStdString() << std::endl;
+  QDir dir(default_root_path + QString(OS_SEP) + "Logs");
+  if (!dir.exists()) {
+    dir.mkpath(".");
+  }
+  std::ofstream file(filename.toStdString(), std::ios::out);
+  file << "Task Name: " << default_task_name.toStdString() << "\n" << std::endl;
+  opt_param.PrintSelf(file);
+  file << "\n" << std::endl;
+  filters_widget->saveAllSettings(file);
+  file.close();
```

Comment:
Consider simplifying how:
* the `Logs` directory is created
* the filename is composed.
* the text is appended to the file by using `QTextStream`

```cpp
  QDir defaultRootDir(default_root_path);
  defaultRootDir.mkpath("Logs");
  QString filename = defaultRootDir.filePath(QString("Logs/%1_%2.log").arg(std::time(nullptr), default_task_name));
  std::cout << "Exporting log file to: " << filename.toStdString() << std::endl;

  QFile file(filename);
  if (!file.open(QIODevice::WriteOnly | QIODevice::Text)) {
    std::cerr << "Failed to open file for writing: " << qPrintable(filename) << std::endl;
    return;
  }

  QTextStream log(&file);

  log << "Task Name: " << default_task_name << "\n";

  xromm::OptimizationParameters& opt_param = tracker->getLatestOptimizationParameters();
  opt_param.PrintSelf(log);
  log << "\n";

  filters_widget->printAllSettings(log);
```

Reply:
I think I am going to continue using `std::ofstream` for writing to the file since there is no conversion to map a `QTextStream` to a `std::ofstream` and linking Qt to libautoscoper (for the `PrintSelf` method) just feels wrong.

Reply:
Agreed, I missed that aspect. Keeping `std::ostream` is then sensible.

I suggest to have something like this then:

```cpp
  QDir defaultRootDir(default_root_path);
  defaultRootDir.mkpath("Logs");
  QString filename = defaultRootDir.filePath(QString("Logs/%1_%2.log").arg(std::time(nullptr), default_task_name));
  std::cout << "Exporting log file to: " << filename.toStdString() << std::endl;

  std::ofstream file(filename.toStdString(), std::ios::out);

  file << "Task Name: " << default_task_name.toStdString() << "\n";
  file << "\n";

  tracker->printLatestOptimizationParameters(file);
  file << "\n";

  filters_widget->printAllSettings(file);
  file << "\n";

  file.close();
```

where a function like `printLatestOptimizationParameters` would be used. For now, I suggest to keep `OptimizationParameters` internal the class and use an API similar across `tracker` and `filters_widget`